napari/_qt/qt_main_window.py

```diff
@@ -384,6 +429,137 @@ def show_notification(notification: Notification):
+        # self._update_version = '0.4.15'
+        # FIXME: To test update options dialog
+        update_info["update"] = True
+        if self._update_version in _settings.updates.update_version_skip:
+            dlg = UpdateStatusDialog(self)
+            dlg.exec_()
+            return
+
+        if update_info["update"]:
+            if running_as_constructor_app():
+                if _settings.updates.update_automatically and startup:
+                    # if automatic, do not prompt a dialog, but start the
+                    # update process right away. Only do this for startup
+                    # Manual checks will not start and automatic update.
+                    self._update_napari()
+                else:
+                    # prompt a dialog for confirmation
+                    dlg = UpdateOptionsDialog(
+                        self, version=self._update_version
+                    )
+                    dlg.exec_()
+
+                    if dlg.is_update():
+                        self._update_napari()
+                    elif dlg.is_update_on_quit():
+                        self._update_on_quit = dlg.is_update_on_quit()
+                        self._update_manager = None
+        else:
+            if not startup:
+                dlg = UpdateStatusDialog(self)
+                dlg.exec_()
+
+    def _update_napari(self):
+        """Launch napari update process and show the notification area."""
+        _settings = get_settings()
+        nightly = _settings.updates.check_nightly_builds
+        self._update_manager = get_update_manager(self)
+        self._update_manager.started.connect(
+            lambda: self._show_activity_dock(True)
+        )
+        self._update_manager.finished.connect(
+            lambda: self._show_activity_dock(False)
+        )
+        self._update_manager.run_update(
+            version=self._update_version,
+            nightly=nightly,
+        )
+        self._update_manager.finished.connect(self._update_finished)
+        self._update_manager.errored.connect(self._update_errored)
+
+    def _update_errored(self, messages):
+        """Run cleanup when the update process failed."""
+        dlg = UpdateErrorDialog(self)
+        dlg.setErrorText(messages)
+        dlg.exec_()
+
+    def _update_finished(self):
+        """Run cleanup actions when the update process finishes."""
+        # FIXME: Needs to show a dialog in case of errors.
+        self._update_on_quit = False
+
+    def _show_activity_dock(self, value):
+        """Show/hide activity dock."""
+        self.statusBar()._toggle_activity_dock(value)
+
+    def check_updates(self, startup=False):
+        """Check for napari available updates.
+
+        Parameters
+        ----------
+        startup: bool
+            This check is running this check on startup. Default is ``False``.
+        """
+        # FIXME: toggle for local testing
+        if is_dev():
+            return
```

Comment:
@DragaDoncila this would need to be uncommented as well.

Also assuming you are inside a conda env.